### ISSUE TITLE
Add the Introductory Popup / Banner (Storybook)

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#8171]
## Feature Description

Create the Introductory Popup / Banner component and add it to Storybook.

Note that this component can build upon that introduced in either https://github.com/google/site-kit-wp/issues/8236 or https://github.com/google/site-kit-wp/issues/8237. Both of these have been added as dependencies, but this issue can be implemented when one of those is merged, whichever comes first.

See [introductory popup / banner](https://docs.google.com/document/d/1MGD5Djy6AeeZC4zBtHqS-lQEWD9jw0kf-IIWw-jLCFU/edit#heading=h.y73fpz30gnsl) in the design doc.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The Introductory Popup / Banner component should be created and stories for it added to Storybook.
* It should be implemented according to the [Figma design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1395-20664&mode=design&t=aTkVS2XahfjwB9Wh-0).
  * Note the variation for the [mobile design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1395-20664&mode=design&t=aTkVS2XahfjwB9Wh-0).
  * ~The tablet design should use the graphic variant with three groups as seen in the [tablet design for the Setup CTA Banner](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=1834-31500&mode=design&t=aTkVS2XahfjwB9Wh-0).~
  * The tablet view should essentially follow the desktop design and layout, as per the tablet [Figma design](https://www.figma.com/file/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?type=design&node-id=3794-17947&mode=design&t=f725NSquEOTsNL3m-0).
    * Note, the need for this amendment was spotted and made during execution.
* The Popup / Banner should have a **Show me** primary CTA and a **Got it** dismissal CTA. However these can remain non-functional, to be hooked up via #8172.

## Implementation Brief

* [x] Add `assets/js/components/OverlayNotification/AudienceSegmentationIntroductoryOverlayNotification.js`
  * You can use `assets/js/components/OverlayNotification/LinkAnalyticsAndAdSenseAccountsOverlayNotification.js` or any other overlay notification as an example
  * Include the CTA's header text and content:
    * Header: `New! Visitor groups`
    * Content: `You can now learn more about your site visitor groups by comparing different metrics`
  * Currently only markup is needed, but you can hook `Got it` CTA with `dismissNotification` action like currently used in other overlay notifications.
  * Use `useBreakpoint` hook, and check the device width to determine which graphic to render. Follow the figma as pointed out in AC, to render appropriate graphics on `BREAKPOINT_DESKTOP` and up, `BREAKPOINT_TABLET` and less up to the `BREAKPOINT_SMALL`, and for `BREAKPOINT_SMALL`.
  * You can include `AUDIENCE_SEGMENTATION_INTRODUCTORY_OVERLAY_NOTIFICATION` constant and pass it to the `setOverlayNotificationToShow` in the `useEffect` hook. Currently it should be shown with only one conditional check -if it has not been dismissed
  * Modify the `isShowingNotification` variable to check for `isShowingOverlayNotification` only if the breakpoint is equal or higher than desktop, otherwise it should only check if notification is not dismissed
  * Wrap the whole markup with main `div.googlesitekit-audience-segmentation-notification`
  * Check if `AudienceSegmentation` feature flag is not enabled to return early
* [ ] Add `assets/sass/components/overlay-notification/_googlesitekit-audience-segmentation-notification.scss`
  * Use the main `googlesitekit-audience-segmentation-notification` class to override the styles, currently what I saw the font weight and letter-spacing styles of introductory popup body text differs from the ones used in overlay notification. And any other differences as per figma design linked in AC
  * When screen size is less then desktop override the display to match figma designs for tablet and mobile respectively, and it should not have `fixed` position 
* This way when this type of banners are hooked to the dashboard in one of the future issues, it can be included in `assets/js/components/OverlayNotification/OverlayNotificationsRenderer.js` for breakpoints less then `BREAKPOINT_DESKTOP` and in `assets/js/components/Header.js` for breakpoints equal or greater than `BREAKPOINT_DESKTOP`

### Test Coverage

* Add stories file for the `AudienceSegmentationIntroductoryOverlayNotification` component

## QA Brief

* Check the copy and design of the widget in Storybook. Check for different viewport and reassess with the Figma design. 
* The notification can also be checked in plugin by enabling the `audienceSegmentation` feature flag. 
* Check the position of the overlay in Tablet and desktop viewports on RTL layout (by Changing WP Language to an RTL language). 
    * It should be placed on the side apposite to the WP Admin Sidebar and not go under the sidebar. 
    * See [this](https://10up.slack.com/archives/CBKKQEBR9/p1712576408439219) slack message for additional details. 

## Changelog entry

* Add introductory popup for Audience Segmentation.

**Comment by tofumatt** [comment]
IB ✅

**Comment by kelvinballoo** [comment]
### QA Update ❌ 

Tested on WP 6.5.2 and PHP 8.0 on develop branch.
I've detailed the issues/clarifications required below.

**ITEM 1:**
The QAB states:  "It should be placed on the side apposite to the WP Admin Sidebar and not go under the sidebar."
What do we mean by not going under the sidebar?
______________________________________________________________________________

**ITEM 2:**
The position of the pop up on desktop isn't correct.
The implementation shows equal margin on the right side and bottom side. But according to figma, the distance is bigger on the right hand side.

Please help to double check the same for tablet if required.

<Details>
Notice that on figma, the right side is longer. Unless, the bottom on the figma is actually not the screen bottom:
<img width="421" alt="Screenshot 2024-04-26 at 12 14 39" src="https://github.com/google/site-kit-wp/assets/125576926/7897ac5d-38f0-4bc0-a2fc-4129eb90a3f5">

Either way, we still need to work on the position because when I enable the audiencesegmentation flag, you will notice that  the popup aligns very closely to the back element (which is not the case on figma):
<img width="442" alt="Screenshot 2024-04-26 at 12 17 29" src="https://github.com/google/site-kit-wp/assets/125576926/9b2f8797-17a3-48c1-b4fe-83b674616ae5">


</Details>
______________________________________________________________________________

**ITEM 3:**
Question: Do we use the font 'Google Sans Display' and 'Google Sans text' be used interchangeably?
The reason I ask is because sometimes I don't see them aligned:

- For the CTA buttons on desktop figma, it's supposed to be 'Google Sans Display'. 

   Implementation is currently 'Google Sans Text'

   <Details>
   Figma:
   <img width="515" alt="Screenshot 2024-04-26 at 12 22 13" src="https://github.com/google/site-kit-wp/assets/125576926/09dd7e2b-7ca1-4ea4-b089-8d08f6c20a91">

   Implementation:
   <img width="387" alt="Screenshot 2024-04-26 at 12 22 32" src="https://github.com/google/site-kit-wp/assets/125576926/58f2c74f-8d1a-47ed-b06d-e9111bba862a">

</Details>

- On mobile the Title implemented is 'Google Sans display' but in figma it's 'Google sans Text'.
   <Details>
   Implementation:
      <img width="502" alt="Screenshot 2024-04-26 at 12 39 17" src="https://github.com/google/site-kit-wp/assets/125576926/280771ce-22c3-4f0f-8670-a3dbfac11a45">

   Figma:
   <img width="839" alt="Screenshot 2024-04-26 at 12 40 07" src="https://github.com/google/site-kit-wp/assets/125576926/d2726ef4-1f59-4ec0-8960-a1b5b4213797">


</Details>
______________________________________________________________________________

**ITEM 4:**
On figma, the CTA text margin on top and bottom is '10px'
Current implementation is 8px. 

<Details>
<img width="325" alt="Screenshot 2024-04-26 at 12 25 39" src="https://github.com/google/site-kit-wp/assets/125576926/960ea5ca-5bab-47e3-9af8-c3b332788eb2">

</Details>
______________________________________________________________________________

**ITEM 5:**
The following actually do not warrant a fix because they are very minor and might not even be noticeable but I am raising them anyway in case we want to be pixel perfect with the designs.

The green colour used is supposed to be B8E5CA instead of B8E6CA.
<Details>
<img width="571" alt="Screenshot 2024-04-26 at 12 29 27" src="https://github.com/google/site-kit-wp/assets/125576926/444fc605-aca2-48a4-abe4-0d91deb93ca0">

</Details>

On figma, the following margins are 25px and current implementation is 24px.
While I think 24px should be the norm, I'm just raising it because there is a deviation from figma.
<Details>
<img width="332" alt="Screenshot 2024-04-26 at 12 31 55" src="https://github.com/google/site-kit-wp/assets/125576926/34a8eca0-db44-49ac-b105-201489740d49">
</Details>

______________________________________________________________________________

**ITEM 6:**
The image on mobile is 267px in width currently.
Based on figma, it should be 295px.
2nd thing to note is that on figma, the image stretches to the bottom of the card and there is no white line below it.

<Details>
Implementation:
<img width="626" alt="Screenshot 2024-04-26 at 11 49 37" src="https://github.com/google/site-kit-wp/assets/125576926/cedc3fdb-6a57-4eb8-a57f-39b36ef3b726">

Figma:
<img width="976" alt="Screenshot 2024-04-26 at 12 42 22" src="https://github.com/google/site-kit-wp/assets/125576926/7054f132-426e-4f0c-8f34-dcb8ca5cf536">


</Details> 

______________________________________________________________________________

**ITEM 7:**
These are all margin/padding issues on mobile.
The following distances are all 20px on figma but on implementation, it's 16px:
<Details>
Figma:
<img width="703" alt="Screenshot 2024-04-26 at 12 50 00" src="https://github.com/google/site-kit-wp/assets/125576926/d48179d6-2007-4d2d-ac41-44cf62d868ec">

Implementation:
<img width="904" alt="Screenshot 2024-04-26 at 12 51 48" src="https://github.com/google/site-kit-wp/assets/125576926/07c9a5e3-b19a-419d-a7ff-9c05961c95f3">

</Details>

The distances on the sides should be 16px based on figma but it's currently larger than that.

<Details>
Figma:
<img width="689" alt="Screenshot 2024-04-26 at 12 53 17" src="https://github.com/google/site-kit-wp/assets/125576926/20f15527-9ba8-4f6d-a049-d2b40e5936d1">

Implementation:
<img width="529" alt="Screenshot 2024-04-26 at 12 54 57" src="https://github.com/google/site-kit-wp/assets/125576926/bf292785-5174-4d06-a3d0-f10f3a74a1f9">


</Details>

There should be more space below the buttons and the image (27px).
Implementation is currently 16px.

<Details>
Figma:
<img width="664" alt="Screenshot 2024-04-26 at 12 56 33" src="https://github.com/google/site-kit-wp/assets/125576926/9859491b-575b-4f2d-9bcf-7d108ea7d524">

Implementation:
<img width="849" alt="Screenshot 2024-04-26 at 12 58 06" src="https://github.com/google/site-kit-wp/assets/125576926/1c4833fc-6626-47e3-9b10-201d316246ef">


</Details>

**Comment by kuasha420** [comment]
Thanks @kelvinballoo. I'll try to address your observations. I'll address whatever needs addressing in a follow up PR. 

1. Sorry for not being somewhat vague here. Please see this [slack message](https://10up.slack.com/archives/CBKKQEBR9/p1714119925664859) for additional context. 
2. Nice find. Looks like the margin is 32px on both right and bottom on tablet, but 42px right, 22px bottom on desktop. I'll spin up a PR to update this. 
3. Can not exactly answer your question about interchangeably using the font, but the examples you've listed are consistent with rest of Site Kit, the CTA is a reusable button used everywhere, and the h3 tag also have same font applied throughout the plugin. I don't think we need to update this. 
4. This is the same as above, the CTA button is a reusable component and consistent across the plugin. 
5. Interesting observations! The `B8E6CA` is coming straight from the exported SVG from Figma. So I'm not sure why the svg and the UI is different. I've exported the file again, and the color comes out as `B8E6CA`. Which is the case for all the overlay images. This is fine. The margin difference is fine as well as we're using standard SK gap values. 
6. That is valid, I'll fix it up in follow up PR. 
7. Interesting observations!
    a. For padding around the widget, I've matched it with other CTA/Banner/Widget margin/padding on mobile for consistent look with our defined gaps. I think they should remain 16px but will ask in the channel to confirm. 
    b. For margins on the side, can you check if the margins matches with other widgets and Banners on SK Dashboard? I remember matching that with the rest of the widget/CTA.
    c. I've intentionally kept it 16px, following our `gaps` elsewhere in the plugin. It should either remain 16 or be 24 px, 27 seems like an odd choice. I'll ask about this one in slack as well and confirm. 

Thank you!

**Comment by techanvil** [comment]
Hey @kuasha420, reviewing the points in your reply above:

> 1. Sorry for not being somewhat vague here. Please see this [slack message](https://10up.slack.com/archives/CBKKQEBR9/p1714119925664859) for additional context. 

Tbh, I am actually a bit confused here - I'm not clear on what the additional context should be taken from the linked thread.

Furthermore though, in theory we shouldn't be integrating this popup/banner into Site Kit as part of this issue. The component is supposed to be shown conditionally upon setting up the Audience Segmentation feature, just adding it to Storybook in this issue and then integrating it via https://github.com/google/site-kit-wp/issues/8172.

Still, I suppose it's behind the feature flag and so it doesn't really hurt to show it unconditionally for now so we can verify it works as expected when integrated, and we can simply apply the relevant conditions as part of 8172.

Anyhow, @kuasha420, please can you double check the linked thread for relevancy and add a bit of further clarification for this point?

> 2.
> ...
> 6.

Your replies to points 2 through 6 LGTM.

> 7. Interesting observations!  
>     a. For padding around the widget, I've matched it with other CTA/Banner/Widget margin/padding on mobile for consistent look with our defined gaps. I think they should remain 16px but will ask in the channel to confirm.  
>     b. For margins on the side, can you check if the margins matches with other widgets and Banners on SK Dashboard? I remember matching that with the rest of the widget/CTA.  
>     c. I've intentionally kept it 16px, following our `gaps` elsewhere in the plugin. It should either remain 16 or be 24 px, 27 seems like an odd choice. I'll ask about this one in slack as well and confirm. 

As you've pointed out, we should be aiming for consistency within the plugin. Sometimes Figma is a bit out of alignment with what's already there - it's something we should certainly aim to improve going forward but in the meantime we do have to use our judgement to assess what the concrete implementation should look like.

In this case, it's clear the margins are consistent with other similar banners e.g. the `AnalyticsAndAdSenseAccountsDetectedAsLinkedOverlayNotification` component which can be seen [here](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/components-analyticsandadsenseaccountsdetectedaslinkedoverlaynotification--default) in Storybook, as well as with the margins surrounding other widgets which use the standard 16/24px gap size. I.e., the widgets should be aligned as illustrated here, and the top margin should use the same value:

![image](https://github.com/google/site-kit-wp/assets/18395600/4559055e-759b-45db-8a6c-74144cee02b8)

So, to summarise point 7 I would say the implementation LGTM as it stands.

**Comment by kuasha420** [comment]
@techanvil @kelvinballoo The first point of ITEM 6 regarding the size is also fine I think, it maintains the aspect ratio and and changes size based on screen size on mobile, so it will not be exactly same as Figma and most other SVGs in the plugin behaves that way. 

With that said, the follow up PR that fixes the overlay position on tablet/desktop (bottom/right margin) and the whitespace under the SVG on mobile is now in CR. 

Cheers

**Comment by techanvil** [comment]
Thanks @kuasha420. The followup PR is merged, back to you for another look, @kelvinballoo.

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

Thanks @techanvil , @kuasha420 .

Noted on all the points clarified.

Tested the outstanding items as follows:

- **ITEM 1:** ✅ 
   The popup appears on the opposite side of the WP sidebar and for a language with opposite writing direction, it doesn't go under the WP Sidebar.
   
   <Details>
   <img width="1290" alt="Screenshot 2024-05-01 at 15 24 40" src="https://github.com/google/site-kit-wp/assets/125576926/3980e3cd-f44b-4abe-b555-40dcda6781fb">

   </Details>
________________________________________________________________________

- **ITEM 2:** ✅ 
   The position of the pop up on desktop is correct now.
   It's 42px from the right side.
   It's 22px from the bottom side.
   
   <Details>
   <img width="498" alt="Screenshot 2024-05-01 at 15 40 52" src="https://github.com/google/site-kit-wp/assets/125576926/eeb00aa4-d602-4e9e-bf14-2fc6ff1a01ea">

   </Details>

________________________________________________________________________

- **ITEM 6:** ✅ 
   There is no longer the white gap below the image.
   <Details>
   <img width="528" alt="Screenshot 2024-05-01 at 16 09 18" src="https://github.com/google/site-kit-wp/assets/125576926/0d88649c-4ab6-4e6d-b482-24c942cfce52">

   </Details>

These are looking good. 
Moving ticket to Approval.
